src/main/java/mate/academy/Dao/BookDaoImpl.java

```diff
@@ -0,0 +1,118 @@
+package mate.academy.Dao;
+
+import mate.academy.exception.DataException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.math.BigDecimal;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class BookDaoImpl implements BookDao{
+    private static final int ID_INDEX = 3;
+    private static final int TITLE_PART = 1;
+    private static final int PRICE_PART = 2;
+    private static final int MINIMUM_CHANGES = 1;
+    private static final int INDEX_ID = 1;
+    private static final String INSERT = "INSERT INTO books (title, price) VALUES (?, ?)";
+    private static final String SELECT_BY_ID = "SELECT * FROM books WHERE id = ?";
+    private static final String SELECT_ALL = "SELECT * FROM books";
+    private static final String UPDATE = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_BY_ID = "DELETE FROM books WHERE id = ?";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(INSERT,
+                     Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(TITLE_PART, book.getTitle());
+            statement.setBigDecimal(PRICE_PART, book.getPrice());
+            if (statement.executeUpdate() < MINIMUM_CHANGES) {
```

Comment:
You don't need this check here, if something goes wrong and record is not inserted, you will catch an exception anyway.